crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -383,6 +383,15 @@ impl ThreadSummaryStatus {
     fn is_unknown(&self) -> bool {
         matches!(self, ThreadSummaryStatus::Unknown)
     }
+
+    /// Transforms the [`ThreadSummaryStatus`] into an optional thread summary,
+    /// for cases where we don't care about distinguishing unknown and none.
+    pub fn maybe_summary(&self) -> Option<&ThreadSummary> {
```

Comment:
I would name this method `summary` instead of `maybe_summary`. It feels a bit more Rust idiomatic, and consistent with the rest of this code base. Not a blocker though.